src/Init/Data/Nat/Lemmas.lean

```diff
@@ -679,6 +679,42 @@ theorem add_mod (a b n : Nat) : (a + b) % n = ((a % n) + (b % n)) % n := by
 @[simp] theorem mod_mul_mod {a b c : Nat} : (a % c * b) % c = a * b % c := by
   rw [mul_mod, mod_mod, ← mul_mod]
 
+@[simp]
+theorem mod_eq_sub_of_le_of_lt {x w n : Nat} (x_ge : w * n ≤ x) (x_lt : x < w * (n + 1)) :
```

Comment:
This feels like the kind of lemma where `calc` mode would help improve clarity, so let's perhaps pair on this tomorrow.

Reply:
Please refactor this proof as
```
theorem mod_eq_sub (x w : Nat) : x % w = x - w * (x / w) := by
  conv => rhs; congr; rw [← mod_add_div x w]
  simp

@[simp]
theorem mod_eq_sub_of_le_of_lt {x w n : Nat} (x_ge : w * n ≤ x) (x_lt : x < w * (n + 1)) :
    x % w = x - w * n := by
  rw [mod_eq_sub]
  congr
  sorry
```
also with a separate lemma to fill in the `sorry` from the inequalities.

(Optional extra: a parallel `Int.mod_eq_sub`.)